compiler/noirc_errors/src/position.rs

```diff
@@ -51,27 +81,49 @@ impl<T> std::borrow::Borrow<T> for Spanned<T> {
     }
 }
 
+#[derive(
+    Copy, Clone, Serialize, Eq, PartialEq, Ord, PartialOrd, Debug, Deserialize, Hash, Default,
+)]
+pub struct SrcId(usize);
+
+impl From<usize> for SrcId {
+    fn from(value: usize) -> Self {
+        Self(value)
+    }
+}
+
+impl From<SrcId> for usize {
+    fn from(val: SrcId) -> Self {
+        val.0
+    }
+}
```

Comment:
What does a `SrcId` represent? There are no comments.

Similarly to my other comments, lets remove the `From` impls here as well. I think they just obfuscate things here. Using methods we can also control the visibility more tightly, e.g. making constructing SrcIds private.

Reply:
documented, impl From<usize> dropped